### ISSUE TITLE
extract object features 

### ISSUE THREAD

**Issue by xxhiao** [TIBHannover/MSVA#10]
Hi, I am trying to extract the object features using your code in https://github.com/VideoAnalysis/EDUVSUM/tree/master/src.

According to your paper, you are using the googleNet trained with imagenet. I assume that you are extracting features using the model "modelInceptionV3" as in the codes. However, the feature shape of " inceptionv3_feature = modelInceptionV3.predict(frmRz299)" is (8,8, 2048). I tried to change the model initialization code to
"modelInceptionV3=InceptionV3(weights='imagenet', pooling='avg', include_top=False)' to get a 2048 feature vector.
However, the object feature vector length is 1024 in the MSVA codes, and I noticed that the values of features from the feature extraction code is quite different from that in the MSVA codes. For the former, the feature values can be larger than 1, but in the latter, the value seemed to be normalized to [0,1] range. 

Have I missed something?

**Comment by StevRamos** [comment]
Hi, I have the same question, but I found a code https://github.com/KaiyangZhou/pytorch-vsumm-reinforce#readme. I think they took those processed videos from that repository. I found another repository https://github.com/SinDongHwan/pytorch-vsumm-reinforce/blob/master/utils/generate_dataset.py in which they tried to replicate the processed dataset (h5 files). I relied on the latter to try and replicate the data sets, you can find the code here https://github.com/StevRamos/video_summarization/tree/main/src. I also compared the shape of the dataset and the ones I made, it seems they are the same but the extracted feature is not the same (I also used google net from pytorch). Let me know if there is another idea that can help us replicate the dataset

**Comment by xxhiao** [comment]
thank you very much, stevRamos. That's very helpful.

By the way, does anyone know know which optical flow algorithm that the authors of MSVA used to extract I3D features?

**Comment by mpalaourg** [comment]
@StevRamos  Our methods for replicating the dataset are very similar (almost identical). I think that the extracted features are not the same with the given dataset (or KaiyangZhou's, where I also think they took them), either because of the different versions used (PyTorch etc.) or because the initial dataset (KaiyangZhou's) wasn't produced by using GoogleNet from PyTorch.

@xxhiao I didn't deal with the optical features, so I can't be of any help for the implementation. If I remember correctly the paper for the architecture used (the pretrained I3D (Inflated 3D ConvNet)) was [this one](https://openaccess.thecvf.com/content_cvpr_2017/papers/Carreira_Quo_Vadis_Action_CVPR_2017_paper.pdf) , where they say on 2.5 
> We computed optical flow with a TV-L1 algorithm.

I think this is a good enough assumption, for the algorithm used.

**Comment by StevRamos** [comment]
@mpalaourg I almost made a big mistake in the Summe dataset. Thanks! Oh I get it. Did you try using other datasets? I am trying to use "CoSum" and "VSUMM", which I have read are also widely used. However, I don't know why they don't use it in this repository. 

I will follow you to see your results when you finish it. Again, thank you very much.
